state-chain/chains/src/sol/instruction_builder.rs

```diff
@@ -269,7 +278,7 @@ impl SolanaInstructionBuilder {
 			VaultProgram::with_id(vault_program)
 				.execute_ccm_native_call(
 					source_chain as u32,
-					source_address.encode(), // TODO: check if this is correct (scale encoding?)
+					Self::destructure_address(source_address),
```

Comment:
@dandanlen This encoding you added (I believe) doesn't work with Solana, it ends up with garbage bytes on chain

Reply:
I don't think I added this - it was already there and I added the TODO comment about whether it should use `encode()` because it didn't make sense to me.